lib/src/upgrader.dart

```diff
@@ -365,7 +365,7 @@ class Upgrader with WidgetsBindingObserver {
         final response = await (iTunes
             .lookupByBundleId(_packageInfo!.packageName, country: country));
 
-        if (response != null) {
+        if (response != null && response['resultCount'] > 0) {
```

Comment:
Another small note. The `upgrader` class does not know the internal details of the `response`. It just passes that to the `ITunesSearchAPI` class, so it would not be a good idea to access the internals of response inside `_updateVersionInfo`.

Reply:
Sorry for the long turn-around time. I took another look and have submitted a different way of solving the issue. In _decodeResults in ITunesSearchAPI the response may not contain any results. So instead of having other classes know about the internal, it seems like this is equivalent to returning null. This also works well since there are checks for whether the response is null or not.